### ISSUE TITLE
User Profile security enhancement

### ISSUE THREAD

**Issue by sumwonlost** [humhub/humhub#3619]
In User Administration -> Settings. I can set Default user profile visibility to “Visible to members+guests“. This make the profile visible to all.

The issue I have is it display ALL profile fields.

It would offer greater security if in Administration -> Users -> Profiles (Manage profile attributes) -> Add new field (same with edit field) you add Public checkbox. If checked the field is visible to both public and members, if unchecked the field is visible to members only.  Again if Visible is unchecked, the field is not visible for both guests and members.

This gives greater control as at to which fields display for guests and which for members.

**Comment by buddh4** [comment]
Yes, I think we need general privacy settings on user level for something like this.
